detection-bot/src/agent.ts

```diff
@@ -0,0 +1,59 @@
+import {
+  Finding,
+  HandleTransaction,
+  TransactionEvent,
+  FindingSeverity,
+  FindingType,
+} from "forta-agent";
+
+import { sharedConstant, sharedConstantsType } from "./shared";
+/**
+ * Wrapper for the handleTransaction
+ * @param sharedConstant : Constants like createAgents function signature and to(polygon) and from(Nethermind) address.
+ * @returns findings: [] || array with alert object
+ */
+export function provideHandleTransaction(
+  sharedConstant: sharedConstantsType
+): HandleTransaction {
+  return async function handleTransaction(
+    txEvent: TransactionEvent
+  ): Promise<Finding[]> {
+    const findings: Finding[] = [];
+    if (
+      txEvent.from.toLowerCase() !=
+        sharedConstant.NETHERMIND_DEPLOYER_ADDRESS.toLowerCase() ||
+      txEvent.to?.toLowerCase() !=
+        sharedConstant.TO_REGISTER_POLYGON_ADDRESS.toLowerCase()
+    ) {
+      return findings;
+    }
+    const listAllRegisterTxEvent = txEvent.filterFunction(
+      sharedConstant.CREATE_AGENT_FUNCTION_SIGNATURE,
+      sharedConstant.TO_REGISTER_POLYGON_ADDRESS
+    );
+    listAllRegisterTxEvent.forEach((newAgentDescription) => {
+      const { owner, metadata, agentId, chainIds } = newAgentDescription.args;
+
+      findings.push(
+        Finding.fromObject({
+          name: "Nethermind Bot Detection",
+          description: "Detects every new Bot deployed from Nethermind Address",
+          alertId: "NEMD-1",
+          severity: FindingSeverity.Info,
+          type: FindingType.Info,
+          metadata: {
```

Comment:
```suggestion
          type: FindingType.Info,
          protocol: "Nethermind"
          metadata: {
```
protocol field is also what we use in a finding object and it's generally name of the client. here we can put "Nethermind"